clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -38,7 +38,6 @@
 #include <cstdio>
```

Comment:
Can we add a test to catch the failure?  How did you observe the failure: some manual run?  Can we turn that into a test?

Reply:
I caught it on the failing test of invariant_checker on a REGDEPS trace, which was failing due to SYSCALL markers that weren't supposed to be there (`type_filter` should have removed them).
Debugging it I realized it was because of `std::numeric_limits<>::min()/max()` on the enums.
I'm not really sure how to make a test to catch this failure, but I could add a check on the `filter_trace_types` and `filter_marker_types` vectors not to be empty (if we enter there, I don't see a reason why they should) https://github.com/DynamoRIO/dynamorio/blob/f63d03d490b6e463e4f77c0545c6dc97e5825a30/clients/drcachesim/tools/filter/record_filter.cpp#L133